get_s3.go

```diff
@@ -205,47 +205,50 @@ func (g *S3Getter) getObject(ctx context.Context, client *s3.S3, dst, bucket, ke
 
 	if g.client != nil && g.client.ProgressListener != nil {
 		fn := filepath.Base(key)
-		body = g.client.ProgressListener.TrackProgress(fn, 0, *resp.ContentLength, resp.Body)
+		body = g.client.ProgressListener.TrackProgress(fn, 0, resp.ContentLength, resp.Body)
 	}
 	defer body.Close()
 
 	// There is no limit set for the size of an object from S3
 	return copyReader(dst, body, 0666, g.client.umask(), 0)
 }
 
-func (g *S3Getter) getAWSConfig(region string, url *url.URL, creds *credentials.Credentials) *aws.Config {
-	conf := &aws.Config{}
+func (g *S3Getter) getAWSConfig(region string, url *url.URL, staticCreds *credentials.StaticCredentialsProvider) (conf aws.Config, err error) {
+	var loadOptions []func(*config.LoadOptions) error
+	var creds aws.CredentialsProvider
+
 	metadataURLOverride := os.Getenv("AWS_METADATA_URL")
-	if creds == nil && metadataURLOverride != "" {
-		creds = credentials.NewChainCredentials(
-			[]credentials.Provider{
-				&credentials.EnvProvider{},
-				&credentials.SharedCredentialsProvider{Filename: "", Profile: ""},
-				&ec2rolecreds.EC2RoleProvider{
-					Client: ec2metadata.New(session.New(&aws.Config{
-						Endpoint: aws.String(metadataURLOverride),
-					})),
-				},
+	if staticCreds == nil && metadataURLOverride != "" {
+		creds = ec2rolecreds.New(func(o *ec2rolecreds.Options) {
+			o.Client = imds.New(imds.Options{
+				Endpoint:          metadataURLOverride,
+				ClientEnableState: imds.ClientEnabled,
 			})
+		})
+	} else if staticCreds != nil {
+		creds = staticCreds
 	}
 
 	if creds != nil {
-		conf.Endpoint = &url.Host
-		conf.S3ForcePathStyle = aws.Bool(true)
-		if url.Scheme == "http" {
-			conf.DisableSSL = aws.Bool(true)
-		}
+		loadOptions = append(loadOptions,
+			config.WithEC2IMDSClientEnableState(imds.ClientEnabled),
+			config.WithCredentialsProvider(creds),
+			config.WithEndpointResolverWithOptions(aws.EndpointResolverWithOptionsFunc(
+				func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+					return aws.Endpoint{URL: url.Host}, nil
+				},
+			)))
 	}
 
 	conf.Credentials = creds
 	if region != "" {
-		conf.Region = aws.String(region)
+		loadOptions = append(loadOptions, config.WithRegion(region))
 	}
 
-	return conf.WithCredentialsChainVerboseErrors(true)
+	return config.LoadDefaultConfig(context.TODO(), loadOptions...)
```

Comment:
This should probably make use of `g.Context()` so that the context is configured correctly

```suggestion
	return config.LoadDefaultConfig(g.Context(), loadOptions...)
```